corehq/ex-submodules/casexml/apps/phone/models.py

```diff
@@ -472,6 +473,41 @@ class IndexTree(DocumentSchema):
     def __repr__(self):
         return json.dumps(self.indices, indent=2)
 
+    @staticmethod
+    def get_all_dependencies(case_id, child_index_tree, extension_index_tree,
```

Comment:
This is the crux of it. Still not convinced that this completely covers the spec (cf. skipped tests).